app/src/androidTest/java/org/mozilla/focus/activity/robots/NotificationRobot.kt

```diff
@@ -31,39 +33,20 @@ class NotificationRobot {
     }
 
     fun expandEraseBrowsingNotification() {
-        val PINCH_PERCENT = 50
-        val PINCH_STEPS = 5
-
-        if (!notificationEraseAndOpenButton.waitForExists(waitingTime)) {
-            if (!notificationTray.ensureFullyVisible(notificationHeader)) {
-                eraseBrowsingNotification.pinchOut(
-                    PINCH_PERCENT,
-                    PINCH_STEPS
-                )
-            } else {
-                notificationHeader.click()
-            }
-            notificationTray.ensureFullyVisible(notificationEraseAndOpenButton)
-            assertTrue(notificationEraseAndOpenButton.exists())
-        }
+        notificationHeader.click()
     }
 
     fun verifySystemNotificationExists(notificationMessage: String) {
-        var notificationFound = false
-
-        do {
-            try {
-                notificationFound = notificationTray.getChildByText(
-                    UiSelector().text(notificationMessage), notificationMessage, true
-                ).waitForExists(waitingTime)
-                assertTrue(notificationFound)
-            } catch (e: UiObjectNotFoundException) {
-                Log.d("TestLog", e.message.toString())
-                // scrolls down the notifications until it reaches the end, then it will close
-                notificationTray.scrollForward()
-                mDevice.waitForIdle()
-            }
-        } while (!notificationFound)
+        val notificationInTray = mDevice.wait(
+            Until.hasObject(
+                By.res(NOTIFICATION_SHADE).hasDescendant(
+                    By.text(notificationMessage)
+                )
+            ),
+            waitingTime
+        )
+
+        assertTrue(notificationInTray)
```

Comment:
I'm still a little concerned that the notification won't always be visible and will need a bit of scrolling to reach it because that's how it started being very flaky in the past, that's why we had this complicated hack. But as we saw, on this new API they behave differently, so let's hope it won't be the case, or we'll fix them as they come.